Fixes span usage for both std::span and cuda::span.

raplonu opened the pull request:
`span` does not provide `cbegin` and `cend` methods among others things. This commit fixes this issue and eases the API restriction regarding const containers.

In addition, `delete_edges` seems to be duplicated. So I get rid of one.

eyalroz commented:
The second delete_edges has different parameters. If you restore it, I think I can accept the PR.

raplonu commented:
In the previous commit `delete_edges` [line 199](https://github.com/eyalroz/cuda-api-wrappers/blob/graph_support/src/cuda/api/graph/template.hpp#L199) was taking a span of non const `edge_t`. But due to the read only usage of the span. I decided to make the span const.

This modification created a redefinition of the one [line 176](https://github.com/eyalroz/cuda-api-wrappers/blob/graph_support/src/cuda/api/graph/template.hpp#L176). That is why I deleted it.

The weird thing is one `delete_edges` calls `cuGraphRemoveDependencies` but the other (the one I removed) calls `cuGraphAddDependencies`.

I can revert the signature of the function but I don't think we want to call `xxxRemovexxx` or `xxxAddxxx` based on the constness of a span.

eyalroz commented:
> The weird thing is one `delete_edges` calls `cuGraphRemoveDependencies` but the other (the one I removed) calls `cuGraphAddDependencies`.
> 
> I can revert the signature of the function but I don't think we want to call `xxxRemovexxx` or `xxxAddxxx` based on the constness of a span.

That sounds like a bug. Anyway, I'll take this PR and work on it further after merging.

eyalroz commented:
But wait, no, I can't make span wholly-const. That would be confusing to users, who might be using it otherwise. And its too inconsistent with std::span / gsl::span. We can't do that.

eyalroz commented:
Please have a look at the graph_support branch now, and let me know if it addresses everything you've brought up here.

raplonu commented:
Hey!

Yep it works without any fix now. 😃

eyalroz commented:
Great, thanks :-)

Also working on memory pool support.
